doc/cluster_design.md

```diff
@@ -0,0 +1,61 @@
+# Design: Clustering in SQLflow to analyze patterns in data
+
+## Concept
+
+For analysts and real business people, in the daily analysis work, most of the work is not prediction, but analysis of the patterns in the data. This can help them mine user behavioral characteristics and differences, helping the business discover value and operate.
+
+This design doc introduces how to support the `Cluster Model` in SQLFlow. 
+
+## User interface
+
+Users usually use a **TRAIN SQL** to train a model in Supervised learning. But, in this scenario, we focus on the extraction of data patterns in unsupervised learning. Therefore, we use **EXTRCT SQL** for pattern extraction, the simple pipeline like:
+
+EXTRCT SQL:
+
+``` sql
+SELECT * FROM train_table
+EXTRCT clusterModel
```

Comment:
"EXTRACT" a model does not read well, we can train a model, save a model, use a model to do prediction but not extract a model. Since the model name is "clusterModel", we can distinguish the model type just like we did in xgboost. Maybe we can use `TRAIN ClusterModel.myDeepClusteringModel` then the code generator can generate specific code for the training pipeline for clustering models.